core/CMakeLists.txt

```diff
@@ -30,7 +25,14 @@ install(
   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 
 install(
-  DIRECTORY include/utils
+  DIRECTORY include/algorithms
   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
```

Comment:
I should've done this myself, but the code is unnecessarily complicated here. There should really only be a single `install(...)` command for the headers. This one:

```cmake
install( DIRECTORY include/
   DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} )
```

I.e. the `/` at the end of `include/` tells CMake to consider the contents of the directory, and not to copy the `include` directory itself as well.

This way we won't have to keep adjusting this technical code whenever a new header sub-directory is added.

Reply:
Good point. Will fix it